### PR TITLE
Add tests for Skeleton templates

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

This PR addresses partially https://github.com/cms-sw/cmssw/issues/29850 by adding tests that include `code-checks`,  `code-format`, and compilation (for selected subsets of the skeletons), following the suggestion in https://github.com/cms-sw/cmssw/issues/29850#issuecomment-1050656903

#### PR validation:

Unit tests run.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@smuzaffar, could you review this one?

File: FWCore/Skeletons/test/runtests.sh [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+# Test output of scripts for which compilation, code-checks, and
+# code-format would be difficult
+pushd TestSubsystem
+mkedlpr TestEDLooper
+mkedlpr TestEDLooperRecData TestRecord TestData1 TestData2
+
+mkesprod TestESProducer
+mkesprod TestESProducerRec TestRecord
+mkesprod TestESProducerRecData TestRecord TestData
+
```
Comment (smuzaffar) [inline FWCore/Skeletons/test/runtests.sh:86]: @makortel , just to match the previous `pushd`, I would suggest to add `popd` here too

**Comment by smuzaffar** [comment]
please test for slc7_aarch64_gcc11

lets tests this for non-x86_64 archs

**Comment by smuzaffar** [comment]
please test for slc7_ppc64le_gcc11

File: FWCore/Skeletons/test/runtests.sh (hunk 1 above)
Comment (makortel) [reply FWCore/Skeletons/test/runtests.sh]: Thanks, added

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@cmsbuild, please test for slc7_aarch64_gcc11

**Comment by makortel** [comment]
@cmsbuild, please test for slc7_ppc64le_gcc11

**Comment by makortel** [comment]
@cmsbuild, please test

x86 tests appear to get lost

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@smuzaffar Do you have any further comments?

File: FWCore/Skeletons/test/runtests.sh [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+popd
+git add TestSubsystem
+git commit -a -m "Add skeleton modules that can be easily compiled"
+scram b code-checks-all
+scram b code-format-all
+git diff
+if [ $(git diff --name-only | grep TestSubsystem | wc -l) -gt 0 ] ; then
+    echo "code-checks or code-format caused differences, skeleton templates need to be updated!"
+    exit 1
+fi
```
Comment (smuzaffar) [inline FWCore/Skeletons/test/runtests.sh:52]: @makortel , may be we can remove these lines to run the tests bit faster ( we are running these any way later in the scipt).

Comment (makortel) [reply FWCore/Skeletons/test/runtests.sh:52]: Do we trust that if the files before `code-{checks,format}-all` compile, they will compile also afterwards? Maybe the risk is negligible, and these are anyway examples and thus not critical.

Comment (smuzaffar) [reply FWCore/Skeletons/test/runtests.sh]: Pre-condition of code-checks/clang-tidy  is that code should compile. So yes we should make the first set of generated filed build error free.

Comment (makortel) [reply FWCore/Skeletons/test/runtests.sh]: Removed now

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@smuzaffar Anything else?

**Comment by smuzaffar** [comment]
No @makortel all look good now

**Comment by makortel** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**